subsys/bluetooth/audio/ascs.c

```diff
@@ -1348,6 +1391,12 @@ static bool ascs_parse_metadata(struct bt_data *data, void *user_data)
 		return false;
 	}
 
+	if (!ascs_valid_metadata_type(data->type, data->data_len)) {
+		result->err = -EINVAL;
+
+		return false;
+	}
```

Comment:
I've hit this one while testing against Android

Reply:
This is true as of today, but tomorrow a new assigned number might be added which Zephyr do not know about. The question is then how this shall be handled. It is my impression that Android assumes a device will ignore metadata it does not need - which is valid to some extend as it would take ages to setup a stream if Android needed to pop metadata one item at the time until a peripheral was happy and accepts the enable request. 
It might be worth doing an errata on this topic to let a peripheral expose its supported metadata types. And in the meantime simply ignore unknown metadata types. For now they are all informative.

Reply:
> This is true as of today, but tomorrow a new assigned number might be added which Zephyr do not know about. 

But can't the same be said about any assigned number with RFU values? Or even other fields with RFU values? 

I think we can all agree that we should reject invalid opcodes in ASCS, and if a new opcode is added to ASCS we simply won't support that until Zephyr has been updated. 

Assigned numbers may be a little different as that is a separate, and informative only, specification. However, from my previous comment (https://github.com/zephyrproject-rtos/zephyr/pull/44522#discussion_r889052788), there is actually a test case to reject invalid metadata based on RFU values, so if we want to be qualifiable we need to be able to reject it, but it is possible to move this rejection to the application layer instead, if you guys think that is better?